### ISSUE TITLE
Error Occurs with a photo whose name is 32 characters or more

### ISSUE THREAD

**Issue by HRK-OSN** [ClosedXML/ClosedXML#1671]
## Read and complete the full issue template

Do not randomly delete sections. They are here for a reason.

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature
- [ ] Question

**Did you test against the latest CI build?**
- [ ] Yes
- [x] No

If you answered `No`, please test with [the latest development build](https://ci.appveyor.com/project/ClosedXML/ClosedXML/branch/develop/artifacts) first.

**Version of ClosedXML**

0.95.4

**What is the current behavior?**

An error occurs when reading an Excel file with a photo whose name is 32 characters or more with ClosedXML.
There are no such restrictions on the ECMA or Excel specifications.

**What is the expected behavior or new feature?**

Please do no verify the attribute value.

**Is this a regression from the previous version?**

Sorry I don't know

## Reproducibility
**This is an important section. Read it carefully. Failure to do so will cause a 'RTFM' comment.**

Without a code sample, it is unlikely that your issue will get attention. Don't be lazy. Do the effort and assist the developers to reproduce your problem. Code samples should be [minimal complete and verifiable](https://stackoverflow.com/help/mcve). Sample spreadsheets should be attached whenever applicable. Remove sensitive information.

**Code to reproduce problem:**
```c#
public void Main()
{
   using (var inputStream = new FileStream("filePath", FileMode.Open)
   using (var workbook = new XLWorkbook(inputStream, XLEventTracking.Disabled)) // Error Occurs
   {
     // .... Some process
   }
}
```

- [x] I attached a sample spreadsheet.  (You can drag files on to this issue)
[Test.xlsx](https://github.com/ClosedXML/ClosedXML/files/6547005/Test.xlsx)

**Comment by Pankraty** [comment]
We are validating picture name with these rules:

1. Picture names cannot be empty
2. Picture names cannot contain any of the following characters: :\/?*[]
3. Picture names cannot be more than 31 characters

They all do not seem to be valid, at least for opening a file.
1. Excel opens such files with no issues and assigns a default name (e.g. Picture 1). Libre Office is fine with having it empty.
2. Neither Excel nor LibreOffice object against naming a picture `:\/?*[]` via UI, and they load the file with such picture with no issues too.
3. Though Excel does not allow to set a name longer than 32 characters (why do we use 31?..) it has nothing against loading a file with longer names for pictures. LibreOffice saves pretty long values effortlessly, and they can be opened in Excel just fine:
![image](https://user-images.githubusercontent.com/19576939/119998776-145bb280-bfe2-11eb-85f7-d9f0e1365352.png)

@Igitur do you have a clue why we validate picture names like this? Shall we remove these validations completely?

**Comment by igitur** [comment]
> @igitur do you have a clue why we validate picture names like this? Shall we remove these validations completely?

I can't say I remember exactly. I might have been that files with long picture names were considered corrupt in earlier Excel versions. Or I tried to replicate what the Excel UI does. Or I copied the validation rules from the sheet names. 

Anyway, I agree that we should remove the validation rules at least for loading existing content. Do you think we should relax the rule and always allow long picture names (hence breaking away from the Excel UI)?

**Comment by Pankraty** [comment]
Yes, I think so. It does not seem this will break something.

**Comment by WEBhicham** [comment]
Hello,

Is there any news/ETA on this?  It is a bit of an annoying bug when dealing with customers excel files. 👼

**Comment by hallsbyra** [comment]
Agree. I too came across this bug today when processing a file that had been modified manually by the customer.

**Comment by erxbout** [comment]
Hello,

I came across this while searching for a solution for "Picture names cannot be empty"
While searching I found this unresolved issue.. the question is now what is planned to be done about this?
My problem was that I (no matter what I did in Onlyoffice or Excel itself to name it) could not load the workbook containing a company logo because the name was empty..
Temporary solution for me was to manually edit the xml reference for each template file we had with pictures in it..

the code that apply these rules as stated above from @Pankraty is implemented as followed
``` C#
        internal void SetName(string value)
        {
            if (String.IsNullOrWhiteSpace(value))
                throw new ArgumentException("Picture names cannot be empty");

            if (value.IndexOfAny(InvalidNameChars.ToCharArray()) != -1)
                throw new ArgumentException($"Picture names cannot contain any of the following characters: {InvalidNameChars}");

            if (value.Length > 31)
                throw new ArgumentException("Picture names cannot be more than 31 characters");

            _name = value;
        }
```

Question is: Is any validation needed except for InvalidNameChars?
Removing this validation would fix this issue and mine and would not be much impact as onlyoffice and excel are both fine with my template before and after my xml change (so the name does not really matter that much so why make matters more complicated than they need to be :D)

wish you a nice day ;)
I could promote a change via PR if we come to some kind of agreement ;)

**Comment by jahav** [comment]
31 character limit was removed in #2836.
